app/models/smart_proxy.rb

```diff
@@ -88,12 +86,19 @@ def has_feature?(feature)
     self.features.any? { |proxy_feature| proxy_feature.name == feature }
   end
 
-  private
-
-  def proxy_status
-    @proxy_status ||= ProxyStatus.new(self)
+  def status
+    @status ||= { :version => ProxyStatus::Version.new(self) }
+    features.map(&:name).each do |feature|
```

Comment:
I'm not following why you do that (or why github keeps removing my comment about this).